src/zkevm_specs/evm/typing.py

```diff
@@ -817,23 +819,25 @@ def _append_steps(
                 exponent = exponent - 1
 
     def _append_padding_row(self, identifier: IntOrFQ):
```

Comment:
Based on the new approach we are following, the dummy rows of `1 ^ 1 == 1` will be after _all_ the exponentiation events. If you look closely to the `add_event` method, after every event, the `_add_padding_row` is called (which is not what we want). So some refactoring is required so that you have a public method `fill_dummy_events` which does this padding and is called from [here](https://github.com/privacy-scaling-explorations/zkevm-specs/blob/735cc398aeecdc08d746676db367437cefaef3b4/tests/evm/test_exp.py#L72):
```python
exp_circuit = ExpCircuit().add_event(base, exponent, randomness, rw_dict.rw_counter).fill_dummy_events()
```

I will let you decide on how you wish to name those methods :)